crates/bevy_transform/src/hierarchy/child_builder.rs

```diff
@@ -471,4 +475,11 @@ mod tests {
             PreviousParent(parent)
         );
     }
+
+    #[test]
+    fn regression_push_children() {
```

Comment:
A more descriptive name would be nice, e.g. `regression_push_same_archetype` (if that is the cause of the issue)